client/src/pages/lessons/edit.tsx

```diff
@@ -326,6 +339,26 @@ const LessonEdit = (props: {
     navigate(`/lessons`);
   }
 
+  function parseProps(
+    props: Array<{ name: string; value: string }>,
+    key: string
+  ): string {
+    return props.find((p) => p.name === key)?.value ?? "";
+  }
+
+  function updateProps(
+    props: Array<{ name: string; value: string }>,
+    key: string,
+    value: string
+  ): Array<{ name: string; value: string }> {
+    for (let i = 0; i < props.length; i++) {
```

Comment:
this is doing a mutation on the props, and would be better/safer to not. Also this will not set a prop unless it already exists. Let's do this instead:

```typescript
interface Prop {
  name: string;
  value: string
}
function copyAndSetProp(props: Array<Prop>, p: Prop): Array<Prop> {
  const pix = props.find(p => p.name == name)
  return (pix >= 0)
    ? props.map((existing, i) => (i === pix) ? p: existing
    : [...props, p]
}
```